braindecode/models/usleep.py

```diff
@@ -229,34 +233,37 @@ def __init__(self,
         # The temporal dimension remains unchanged
         # (except through the AvgPooling which collapses it to 1)
         # The spatial dimension is preserved from the end of the UNet, and is mapped to n_classes
-        self.clf = nn.Sequential(
-            nn.Conv1d(
-                in_channels=channels[1],
-                out_channels=channels[1],
-                kernel_size=1,
-                stride=1,
-                padding=0,
-            ),                         # output is (B, C, 1, S * T)
-            nn.Tanh(),
-            nn.AvgPool1d(input_size),  # output is (B, C, S)
-            nn.Conv1d(
-                in_channels=channels[1],
-                out_channels=n_classes,
-                kernel_size=1,
-                stride=1,
-                padding=0,
-            ),                         # output is (B, n_classes, S)
-            nn.ELU(),
-            nn.Conv1d(
-                in_channels=n_classes,
-                out_channels=n_classes,
-                kernel_size=1,
-                stride=1,
-                padding=0,
-            ),
-            nn.Softmax(dim=1) if apply_softmax else nn.Identity(),
-            # output is (B, n_classes, S)
-        )
+        self.len_last_layer = channels[1] * seq_length * input_size
+        self.return_feats = return_feats
+        if not return_feats:
+            self.clf = nn.Sequential(
+                nn.Conv1d(
+                    in_channels=channels[1],
+                    out_channels=channels[1],
+                    kernel_size=1,
+                    stride=1,
+                    padding=0,
+                ),                         # output is (B, C, 1, S * T)
+                nn.Tanh(),
+                nn.AvgPool1d(input_size),  # output is (B, C, S)
+                nn.Conv1d(
+                    in_channels=channels[1],
+                    out_channels=n_classes,
+                    kernel_size=1,
+                    stride=1,
+                    padding=0,
+                ),                         # output is (B, n_classes, S)
+                nn.ELU(),
+                nn.Conv1d(
+                    in_channels=n_classes,
+                    out_channels=n_classes,
+                    kernel_size=1,
+                    stride=1,
+                    padding=0,
+                ),
```

Comment:
@tgnassou can you explain somewhere why you don't take as features the activations before the last softmax?

Reply:
@agramfort I do it because I take thefeatures before what the author calls classifier (3 conv layers). But it is also possible to consider the features the activation before the last softmax. Do you think there is a better choice ?

Reply:
ok then. Can you just add a comment in the code about this?